### PR TITLE
Add repo url, home and sources to availablepackagedetail

### PR DESCRIPTION
**Pull request by gfichtenholt**
Related to issue: put the new fields in middle of the message, as opposed to the end, figured we still have the freedom to do so at this point. 

Unrelated to issue: also did a little bit of tidying up in flux plug-in trying to use types.NamespacedName where possible as it seems a bit cleaner/easier to read

### CODE REVIEW

**Comment by gfichtenholt** [comment]
couple of more minor changes and it should be ready for review

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go [hunk 1]
```diff
@@ -229,18 +233,24 @@ func (s *Server) installedPackageDetail(ctx context.Context, name, namespace str
...
 		CurrentPkgVersion:     lastAppliedRevision,
 		ValuesApplied:         valuesApplied,
 		ReconciliationOptions: installedPackageReconciliationOptionsFromUnstructured(unstructuredRelease.Object),
 		AvailablePackageRef:   availablePackageRef,
 		// TODO (gfichtenholt) PostInstallationNotes
+		// looks like I may have to resort to getting the notes via helm APIs, flux doesn't do it
+		// see discussion in https://cloud-native.slack.com/archives/CLAJ40HV3/p1629244025187100
+		// bottom line: "helm get notes" is a read-only activity that you can replicate in your
+		// own code by copying the implementation, or calling to the library function if there is one
+		// https://github.com/helm/helm/blob/main/cmd/helm/get_notes.go.
```
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:252]: You can use the helm action in code the same way we do in the helm plugin, I would think. (Not necessary now, just related to your comment :) )

**Review by absoludity — APPROVED** [review]
> Related to issue: put the new fields in middle of the message, as opposed to the end, figured we still have the freedom to do so at this point.

Yep, +1

> 
> Unrelated to issue: also did a little bit of tidying up in flux plug-in trying to use types.NamespacedName where possible as it seems a bit cleaner/easier to read

Nice, also nice cleanup of the tests to re-use (and abstract) the test fixture setups.

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/chart_test.go [hunk 2]
```diff
@@ -320,12 +342,13 @@ func TestNegativeGetAvailablePackageDetail(t *testing.T) {
 	}
 }
 
-func TestNegativeGetAvailablePackageDetailNonExistingRepoOrInvalidPkgVersion(t *testing.T) {
+// TODO: (gfichtenholt) some, if not all, these scenarios probably ought to return
+// codes.NotFound instead of codes.Internal. The spec does not specify yet.
```
[the remark this answers is not in the document]
Comment (gfichtenholt) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/chart_test.go:346]: no technical reason, just parity with what helm plug-in does. Unless I am grossly mistaken the only place that returns NotFound is one path in GetInstalledPackageDetail. We ought to open a separate issue to address this consistently across plug-ins.

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go [hunk 3]
```diff
@@ -49,26 +50,28 @@ func (s *Server) listReleasesInCluster(ctx context.Context, namespace string) (*
 		Resource: fluxHelmReleases,
 	}
 
-	if releases, err := client.Resource(releasesResource).Namespace(namespace).List(ctx, metav1.ListOptions{}); err != nil {
+	return client.Resource(releasesResource).Namespace(namespace), nil
+}
+
+// namespace maybe "", in which case repositories from all namespaces are returned
```
[the remark this answers is not in the document]
Comment (gfichtenholt) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:56]: yep

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go [hunk 4]
```diff
@@ -144,17 +147,18 @@ func (s *Server) installedPkgSummaryFromRelease(unstructuredRelease map[string]i
 		// e.g. "default-my-nginx". The spec somewhat vaguely states "The name of the chart as made available
 		// by the HelmRepository (without any aliases), for example: podinfo". So, we can't exactly do a "get"
 		// on the name, but have to iterate the complete list of available charts for a match
-		url, err := findUrlForChartInList(chartsFromCluster, repoName, chartName, chartVersion)
-		if err == nil && url != "" {
+		tarUrl, err := findUrlForChartInList(chartsFromCluster, repoName, chartName, chartVersion)
+		if err == nil && tarUrl != "" {
 			chartID := fmt.Sprintf("%s/%s", repoName, chartName)
-			pkgDetail, _ = availablePackageDetailFromTarball(chartID, url)
+			pkgDetail, _ = availablePackageDetailFromTarball(chartID, tarUrl)
```
[the remark this answers is not in the document]
Comment (gfichtenholt) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:153]: yeah, not sure why I did this at the time. I will fix this

File: cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go (hunk 1 above)
Comment (gfichtenholt) [reply cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go:252]: yes, I plan on it

**Pull request merged**